src/lib/Export/Input/CommandInputResolver.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Personalization\Export\Input;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\InputInterface;
+
+final class CommandInputResolver implements CommandInputResolverInterface
+{
+    public function resolve(InputInterface $input, ?Application $application = null): array
+    {
+        $inputOptions = $input->getOptions();
+        $options = [];
+        $inputOptions =
+            null !== $application
+                ? array_diff_key($inputOptions, $application->getDefinition()->getOptions())
```

Comment:
Why do we need to perform this diff? At first glance it seems that this will not do what you're expecting it to, i.e. not every entry from the expected resulting array will be present (and, judging from the declaration on the interface, you expect them to).

Needs test to prove that this works.

My second thought here is that this class seems to be unnecessary. You're trying to transform the options coming from Command definition, that usually should be already applied (i.e. Command should provide you with defaults for all the options). So in theory you should only need to map the keys to the format expected in the Export client.

Am I correct?

Reply:
This diff provides only filtered options that are used in export and next transform keys to expected format for those options. 

I added test that explain how it works: https://github.com/ezsystems/ezrecommendation-client/pull/69/commits/e97470f5300edfdd6db3e7a476c91cef9d9adc8b